scholarx/archive/functions.js

```diff
@@ -49,12 +60,46 @@ async function getData() {
 
 async function loadData() {
     const payload = await getData();
-    const mentorProfiles = payload.mentor2021.concat(payload.mentor2020).concat(payload.mentor2019);
+    [mentor[2019],mentor[2020],mentor[2021]] = [payload.mentor2019,payload.mentor2020,payload.mentor2021];
+    [mentee[2019],mentee[2020],mentee[2021]] = [payload.mentee2019,payload.mentee2020,payload.mentee2021];
+ 
+    const mentorProfiles = mentor[2019].concat(mentor[2020]).concat(mentor[2021]);
     let mentor_Profiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfiles });
     $("#mentorProfiles").html(mentor_Profiles);
 
-    const menteeProfiles = payload.mentee2021.concat(payload.mentee2020).concat(payload.mentee2019);
+    const menteeProfiles = mentee[2019].concat(mentee[2020]).concat(mentee[2021]);
     let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfiles });
     $("#menteeProfiles").html(mentee_Profiles);
 }
 loadData();
+function renderData(mentorYear,menteeYear) {
+    let mentor_Profiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorYear });
+    let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeYear });
+    $("#mentorProfiles").html(mentor_Profiles);
+    $("#menteeProfiles").html(mentee_Profiles);
+}
+function renderAll() {
+    const mentorProfiles = mentor[2019].concat(mentor[2020]).concat(mentor[2021]);
+    const menteeProfiles = mentee[2019].concat(mentee[2020]).concat(mentee[2021]);
+    let mentor_Profiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfiles });
+    let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfiles });
+    $("#mentorProfiles").html(mentor_Profiles);
+    $("#menteeProfiles").html(mentee_Profiles);
+}
+function cohort(){
+    let year = 2019;
+    let mentorsData = [];
+    let menteesData = [];
+    for(let i=0; i<3; i++){
+        if(document.getElementById("chk"+year).checked == true)
+        {
+            mentorsData = mentorsData.concat(mentor[year]),
+            menteesData = menteesData.concat(mentee[year]),
+            renderData(mentorsData,menteesData)
+        }
+        else if(document.getElementById("chk2019").checked == false && document.getElementById("chk2020").checked == false && document.getElementById("chk2021").checked == false){
+            renderAll();
+        }
```

Comment:
This part can be moved outside the loop.
```suggestion
       if(!document.getElementById("2019").checked && !document.getElementById("2020").checked && !document.getElementById("2021").checked){
            renderAll();
        } else {
        // Insert the for loop here
        }
```